reindex/reindex.go

```diff
@@ -37,46 +37,45 @@ type Builder interface {
 	SetPostBuildCallback(func())
 }
 
-// Scheduler periodically updates the Spack buildcache index.
-type Scheduler struct {
-	*Throttler
-	conf    *config.Config
-	builder Builder
+type Reindexer struct {
+	conf *config.Config
+	sync.Mutex
 }
 
-// NewScheduler returns a scheduler that can update the Spack buildcache index
-// periodically.
-func NewScheduler(conf *config.Config, builder Builder) *Scheduler {
-	reindex := func() {
-		cmd := exec.Command(conf.Spack.Path, "buildcache", "update-index", "--", conf.S3.BinaryCache) //nolint:gosec
-		out, err := cmd.CombinedOutput()
+func New(conf *config.Config) *Reindexer {
+	return &Reindexer{conf: conf}
+}
 
-		var outstr, errstr string
+// Reindex runs `spack buildcache update-index` on the configured S3
+// BinaryCache.
+//
+// If a reindex is currently running, queues the reindex until after the current
+// run.
+//
+// Logs when the reindex actually starts running, and when it ends, or if it
+// fails.
+func (r *Reindexer) Reindex() {
+	r.Lock()
```

Comment:
Do we need more that one re-index in the queue?

Would be better to have a flag that only lets a single re-run happen.

Reply:
Yes, we'll re-implement, thanks.